cmd/account/mgmt/account-unassign.go

```diff
@@ -131,54 +127,124 @@ func (o *accountUnassignOptions) run() error {
 	// loop through accounts list and untag and move them back into root OU
 	for _, id := range accountIdList {
 
+		// untag account
 		err = o.untagAccount(id)
 		if err != nil {
 			return err
 		}
 
+		// move account
 		err = o.moveAccount(id, rootID, destinationOU)
 		if err != nil {
 			return err
 		}
+		// instantiate new client with AssumeRole
+		newAWSClient, err := o.assumeRoleForAccount(id)
+		if err != nil {
+			return err
+		}
+		// delete roles
+		err = o.deleteRoles(newAWSClient)
+		if err != nil {
+			return err
+		}
+		// list iam users created by each account and append to slice
+		users, err := o.listUsersFromAccount(newAWSClient, id)
+		if err != nil {
+			return err
+		}
+
+		allUsers = append(allUsers, users...)
+
 	}
 
-	// Delete login profile
-	err = o.deleteLoginProfile(accountUsername)
-	if err != nil {
-		fmt.Println(err)
+	for _, userName := range allUsers {
+
+		// Delete login profile
+		err = o.deleteLoginProfile(userName)
+		if err != nil {
+			fmt.Println(err)
+		}
+		// Delete access keys
+		err = o.deleteAccessKeys(userName)
+		if err != nil {
+			fmt.Println(err)
+		}
+		// Delete signing certificates
+		err = o.deleteSigningCert(userName)
+		if err != nil {
+			fmt.Println(err)
+		}
+		// Delete policies
+		err = o.deletePolicies(userName)
+		if err != nil {
+			fmt.Println(err)
+		}
+		// Delete attached policies
+		err = o.deleteAttachedPolicies(userName)
+		if err != nil {
+			fmt.Println(err)
+		}
+		// Delete groups
+		err = o.deleteGroups(userName)
+		if err != nil {
+			fmt.Println(err)
+		}
+		// Delete user
+		err = o.deleteUser(userName)
+		if err != nil {
+			fmt.Println(err)
+		}
 	}
-	// Delete access keys
-	err = o.deleteAccessKeys(accountUsername)
-	if err != nil {
-		fmt.Println(err)
+
+	return nil
+}
+
+func (o *accountUnassignOptions) assumeRoleForAccount(account_id string) (awsprovider.Client, error) {
+
+	roleArn := fmt.Sprintf("arn:aws:%s:iam::role/OrganizationAccountAccessRole", account_id)
```

Comment:
This ARN is not valid.

It should be `arn:aws:iam::%s/role/OrganizationAccountAccessRole`